src/App.tsx

```diff
@@ -1,24 +1,248 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useMemo,
+  ChangeEvent,
+  FormEvent,
+  useCallback,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo, StatusToFilter } from './types/Todo';
+import {
+  addTodo,
+  getTodos,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { getFilteredTodos } from './helper/GetFilteredTodos';
+import { Loader } from './components/Loader/Loader';
 
-const USER_ID = 0;
+const USER_ID = 7009;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorDescription, setErrorDescription] = useState('');
+  const [isListLoading, setIsListLoading] = useState(true);
+  const [statusToFilter, setStatusToFilter] = useState(StatusToFilter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inputStatus, setInputStatus] = useState(false);
+  const [userText, setUserText] = useState('');
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const [amountOfActiveTodos, amoutOfCompletedTodos] = useMemo(
+    () => [
+      todos.filter(({ completed }) => !completed).length,
+      todos.filter(({ completed }) => completed).length,
+    ],
+    [todos],
+  );
+
+  const getTodosFromServer = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setErrorDescription('Unable to load todos');
+    } finally {
+      setIsListLoading(false);
+    }
+  };
+
+  const addNewTodo = async (title: string) => {
+    setInputStatus(true);
+
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    if (title.trim() === '') {
+      setErrorDescription('Title cant be empty');
+      setInputStatus(false);
+      setUserText('');
+
+      return;
+    }
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    try {
+      const newTempTodo = await addTodo(newTodo);
+
+      setTodos(prev => ([...prev, newTempTodo]));
```

Comment:
not sure if tempTodo should be added to array of todos, read description once again